### ISSUE TITLE
[BUG] - Some cardano-cli queries do not work without "TestEnableDevelopmentNetworkProtocols" set to true

### ISSUE THREAD

**Issue by ArturWieczorek** [input-output-hk/cardano-node#2954]
**Internal**

**Area**
`cardano-cli`

**Summary**

Query:  `cardano-cli query utxo --tx-in TX-IN`

There is an `ShelleyEncoderUnsupportedQuery` error when trying to use `cardano-cli query utxo --tx-in TX-IN` without adding "TestEnableDevelopmentNetworkProtocols" flag set to true in config file.

Before adding `"TestEnableDevelopmentNetworkProtocols"`:

```sh
cardano-cli query utxo --whole-utxo --testnet-magic 3 | grep ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986
ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986     0        1000000 lovelace
ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986     1        5264604 lovelace

cardano-cli query utxo --tx-in ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986#1 
--testnet-magic 3

cardano-cli: ShelleyEncoderUnsupportedQuery (SomeSecond (GetUTxOByTxIn (fromList [TxInCompact (TxId {_unTxId = 
SafeHash "ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986"}) 0]))) ShelleyNodeToClientVersion3
```

After adding `"TestEnableDevelopmentNetworkProtocols": true` to config file:

```sh
cardano-cli query utxo --tx-in ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986#1 
--testnet-magic 3
                           TxHash                                 TxIx        Amount
--------------------------------------------------------------------------------------
ef0e0a63ec342c267c8618d8988edaca361a38820532040502da1b76ebc43986     1        5264604 lovelace
```

**Network**: Shelley QA

```sh
cardano-cli --version
cardano-cli 1.28.0 - linux-x86_64 - ghc-8.10
git rev 48429531f0d3d71fadce9a5971bf56a6df396f2d

cardano-node --version
cardano-node 1.28.0 - linux-x86_64 - ghc-8.10
git rev 48429531f0d3d71fadce9a5971bf56a6df396f2d
```
**Steps to reproduce**

1. start node `1.28.0` for `shelley_qa`:
```
nix-build -A scripts.shelley_qa.node -o shelley_qa-node-local
./shelley_qa-node-local/bin/cardano-node-shelley_qa
```
2. set `CARDANO_NODE_SOCKET_PATH` for `cardano-cli` and issue queries from above example without and with `"TestEnableDevelopmentNetworkProtocols": true` in config file.

**Expected behavior**
`"TestEnableDevelopmentNetworkProtocols": true` should not be needed for queries to work properly.

**Additional context**
N/A

**Comment by ArturWieczorek** [comment]
After checking other available commands for `cardano-cli query` :  
```
protocol-parameters      
tip                      
stake-distribution       
stake-address-info       
utxo                     
ledger-state                                      
protocol-state                                   
stake-snapshot                                    
pool-params          
```
`utxo --tx-in` is the only one that is currently failing.

**Comment by Jimbo4350** [comment]
Fix #2957

**Comment by Jimbo4350** [comment]
This is not a bug. See: https://input-output-rnd.slack.com/archives/GHZ4R8H2B/p1627299165145500?thread_ts=1626793097.127400&cid=GHZ4R8H2B @dorin100

**Comment by dorin100** [comment]
it is a bug for the Alonzo hard fork - my understanding is that all available CLI commands should work after the Alonzo hard fork without any development flag @Jimbo4350

**Comment by dorin100** [comment]
this is fixed in `node-qa-1.0` node tag
